### PR TITLE
Initial contribution of remote tap extension

### PR DESCRIPTION
**Pull request by atoulme**
**Description:**
Adds the basis of a new extension, the remote ~observer~ tap extension, to be used to display the contents of the data observed via remote observer processors.

This PR deals only with the structure and required files related to the extension.

**Link to tracking Issue:**
#19634

**Testing:**
N/A

**Documentation:**
README

### CODE REVIEW

**Comment by rmfitzpatrick** [comment]
Since there are already [observer extensions](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/extension/observer) that have differing functionality, would naming this "websocket extension" and migrating the "websocket processor" to a different package name make sense?

Excited for this component to be added.

**Comment by atoulme** [comment]
Discussed in SIG meeting that indeed observer is already taken, we should take up a new name. "tap" was proposed.

File: extension/remotetapextension/extension.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+		err := s.server.ListenAndServe()
+		if err != nil && !errors.Is(err, http.ErrServerClosed) {
+			host.ReportFatalError(err)
+		}
+	}()
+	return nil
+}
+
+func (s *remoteObserverExtension) Shutdown(_ context.Context) error {
+	return nil
```
Comment (MovieStoreGuy) [inline extension/remotetapextension/extension.go:49]: Not closing the http server?

File: extension/remotetapextension/main/main.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
+// Copyright The OpenTelemetry Authors
```
Comment (MovieStoreGuy) [inline extension/remotetapextension/main/main.go:1]: Is there any reason for a main file for this?

File: extension/remotetapextension/main/main.go [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+	ext, err := f.CreateExtension(context.Background(), cs, f.CreateDefaultConfig())
+	if err != nil {
+		cs.Logger.Fatal(err.Error())
+	}
+	err = ext.Start(context.Background(), componenttest.NewNopHost())
+	if err != nil {
+		cs.Logger.Fatal(err.Error())
+	}
+	c := make(chan os.Signal, 1)
+	signal.Notify(c, os.Interrupt, syscall.SIGTERM)
```
Comment (MovieStoreGuy) [inline extension/remotetapextension/main/main.go:37]: There is a method named `signal.NotifyContext` that does all this for you.

File: extension/remotetapextension/extension.go (hunk 1 above)
Comment (atoulme) [reply extension/remotetapextension/extension.go]: oh yes. Fixing

File: extension/remotetapextension/main/main.go (hunk 2 above)
Comment (atoulme) [reply extension/remotetapextension/main/main.go]: Maybe there was but not attached to it, deleting

File: extension/remotetapextension/README.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+| Stability     | [development]  |
+| Distributions | [] |
+| Issues        | [![Open issues](https://img.shields.io/github/issues-search/open-telemetry/opentelemetry-collector-contrib?query=is%3Aissue%20is%3Aopen%20label%3Aextension%2Fremotetap%20&label=open&color=orange&logo=opentelemetry)](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues?q=is%3Aopen+is%3Aissue+label%3Aextension%2Fremotetap) [![Closed issues](https://img.shields.io/github/issues-search/open-telemetry/opentelemetry-collector-contrib?query=is%3Aissue%20is%3Aclosed%20label%3Aextension%2Fremotetap%20&label=closed&color=blue&logo=opentelemetry)](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues?q=is%3Aclosed+is%3Aissue+label%3Aextension%2Fremotetap) |
+| [Code Owners](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/CONTRIBUTING.md#becoming-a-code-owner)    | [@atoulme](https://www.github.com/atoulme) |
+
+[development]: https://github.com/open-telemetry/opentelemetry-collector#development
+<!-- end autogenerated section -->
+
+This extension runs as a Web server that loads the remote observers that are registered against it.
```
Comment (yurishkuro) [inline extension/remotetapextension/README.md:14]: > loads the remote observers

what does this mean? Is it loading executable code? Could this link to more details about "remote observers"

Comment (atoulme) [reply extension/remotetapextension/README.md:14]: See here: https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/remoteobserverprocessor

Comment (yurishkuro) [reply extension/remotetapextension/README.md:14]: Sorry, that doesn't help. WebSocket processor just streams data out, it doesn't have any observers that "register".

Comment (MovieStoreGuy) [reply extension/remotetapextension/README.md:14]: I believe the intent of the description is:

> This extensions connections to a configured websocket processor to provide a simple web interface for examining running data.

Please correct me if I am wrong here.

**Review by MovieStoreGuy — COMMENTED** [review]
The code makes sense to me, just need a clarification for the README and some tests added and be good to go.

File: extension/remotetapextension/README.md (hunk 4 above)
Comment (atoulme) [reply extension/remotetapextension/README.md:14]: Websocket processor was renamed to remote observer processor. The folder and code was changed but the readme lags behind, sorry.

**Comment by atoulme** [comment]
> The code makes sense to me, just need a clarification for the README and some tests added and be good to go.

What tests are missing?

**Comment by atoulme** [comment]
@MovieStoreGuy let me know what else I should cover. It should be good to go here, bear in mind this is just a first PR to set the skeleton of the component, and we'll have more chats afterwards.

**Review by MovieStoreGuy — APPROVED** [review]

**Pull request merged**